ui/core/src/model/datasource.ts

```diff
@@ -42,10 +42,21 @@ export interface DatasourceSpec<PluginSpec = UnknownSpec> {
 }
 
 /**
- * A selector for pointing at a specific Datasource. If name is omitted, it's assumed that you want the default
- * Datasource for the specified kind.
+ * A selector for pointing at a specific Datasource.
  */
 export interface DatasourceSelector {
+  /**
+   * Kind of the datasource.
+   */
   kind: string;
+  /**
+   * Group of the datasource.
+   * Omit it if you don't store datasource by group.
+   */
+  group?: string;
```

Comment:
what do you mean `store datasource by group` ? Can you explain a bit furthermore ?

Reply:
I mean that for embedded use case, if it is useless for the developer to have different level of datasources, it can just omit it. Then there won't be any grouping of the datasources and the chip in the selector won't be displayed at all. I can elaborate a bit more in the comment.

Reply:
I don't get what you are doing here. The datasource selector represents the way the datasource will be selected based on the info stored in the dashboard.
A group is not reflected in the dashboard data-model and it shouldn't be since we defined a long time ago that only `name` and `kind` will be the parameters used to select the datasource you want to use.

Then the system behind will find the appropriate datasource. We talked also about selecting a special datasource coming from a dedicated group but we decided to not consider it to have something simpler and an override system with no exception.

This shouldn't be altered when changing the display of the datasource dropdown from my opinion.

Reply:
This group field is used in UI only. Ignored by backend. 

In a perfect world, I would do a dedicated data model for the view but it requires a mapper that I didn't manage to put in place because of the architecture of the UI, tightly coupled with the API datamodel. :/ 

Any help from you or @AntoineThebaud  would be appreciated, if we can do so alright!

Reply:
Exactly what we talked about yes.
Good idea to check variables, but the VariableDefinition is actually the type used at the same time for the view and backend request. The difference in variables is that the provider is done differently, with a zustand store. It allows us to decouple definition coming from the backend, and state coming from the view. 

The spirit is completely different in the datasource provider. The solution is probably to integrate zustand in the datasource provider the variable one but it would require a big refactoring, I'm not sure I am skilled enough. 

It makes me think of Steven who was regularly saying "hey guys what about having zustand store for the datasources ?" This is typically the situation where we can say he's right haha

Reply:
@Nexucis @AntoineThebaud Is it ok for you if we leave it like that or no ? Decouple the backend model and view model for the ``DatasourceSelector`` will be a challenge.

I agree that it's not ideal to put a group there knowing that it is considered differently in the backend and frontend, but that would be too much work for this PR. 
I just added a comment 
![image](https://github.com/perses/perses/assets/10258608/faafe5b8-6853-458c-ae8a-78a7f95486c7)

Reply:
I would be ok with that if we are ok and are going to solve the TODO

Reply:
Yes, I can work on that later. As you know, I'm already thinking of adopting a perses panel in our UI, and in that context, I would like to implement a simpler datasource provider for this kind of purpose. Something more agnostic to the backend.

Just to say that I'm interested in all that datasource provider topic and I'm curious to work on that and discover the other benefits to have a zustand store here.